src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -177,6 +178,8 @@ private async Task<IFormCollection> InnerReadFormAsync(CancellationToken cancell
                             throw new InvalidDataException("Form section has invalid Content-Disposition value: " + section.ContentDisposition);
                         }
 
+                        contentTypesAccumulator.Append(contentDisposition.Name.Value, section.ContentType);
```

Comment:
Move this under the IsFormDisposition case. Files already give you access to their headers, and you don't want to track this for 'other' types that we discarded.